projects/packages/masterbar/src/admin-menu/class-atomic-admin-menu.php

```diff
@@ -0,0 +1,591 @@
+
+		// Move "Add New" plugin submenu to the top position.
+		foreach ( $plugins_submenu as $submenu_key => $submenu_keys ) {
+			if ( 'plugin-install.php' === $submenu_keys[2] ) {
+				// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+				$submenu['plugins.php'] = array( $submenu_key => $plugins_submenu[ $submenu_key ] ) + $plugins_submenu;
+			}
+		}
+
+		$submenus_to_update = array( 'plugin-install.php' => $plugins_slug );
+
+		$this->update_submenus( 'plugins.php', $submenus_to_update );
+	}
+
+	/**
+	 * Adds the site switcher link if user has more than one site.
+	 */
+	public function add_browse_sites_link() {
+		$site_count = get_user_option( 'wpcom_site_count' );
+		if ( ! $site_count || $site_count < 2 ) {
+			return;
+		}
+
+		// Add the menu item.
+		// @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. TODO add link with Trac issue.
+		add_menu_page( __( 'site-switcher', 'jetpack-masterbar' ), __( 'Browse sites', 'jetpack-masterbar' ), 'read', 'https://wordpress.com/sites', null, 'dashicons-arrow-left-alt2', 0 );
+		add_filter( 'add_menu_classes', array( $this, 'set_browse_sites_link_class' ) );
+	}
+
+	/**
+	 * Adds a custom element class for Site Switcher menu item.
+	 *
+	 * @param array $menu Associative array of administration menu items.
+	 *
+	 * @return array
+	 */
+	public function set_browse_sites_link_class( array $menu ) {
+		foreach ( $menu as $key => $menu_item ) {
+			if ( 'site-switcher' !== $menu_item[3] ) {
+				continue;
+			}
+
+			$menu[ $key ][4] = add_cssclass( 'site-switcher', $menu_item[4] );
+			break;
+		}
+
+		return $menu;
+	}
+
+	/**
+	 * Adds a link to the menu to create a new site.
+	 */
+	public function add_new_site_link() {
+		$site_count = get_user_option( 'wpcom_site_count' );
+		if ( $site_count && $site_count > 1 ) {
+			return;
+		}
+
+		// @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. TODO add link with Trac issue.
+		add_menu_page( __( 'Add New Site', 'jetpack-masterbar' ), __( 'Add New Site', 'jetpack-masterbar' ), 'read', 'https://wordpress.com/start?ref=calypso-sidebar', null, 'dashicons-plus-alt' );
+	}
+
+	/**
+	 * Adds site card component.
+	 */
+	public function add_site_card_menu() {
+		$default        = plugins_url( 'globe-icon.svg', __FILE__ );
+		$icon           = get_site_icon_url( 32, $default );
+		$blog_name      = get_option( 'blogname' ) !== '' ? get_option( 'blogname' ) : $this->domain;
+		$is_coming_soon = ( new Status() )->is_coming_soon();
+
+		$badge = '';
+
+		if ( get_option( 'wpcom_is_staging_site' ) ) {
+			$badge .= '<span class="site__badge site__badge-staging">' . esc_html__( 'Staging', 'jetpack-masterbar' ) . '</span>';
+		}
+
+		// @phan-suppress-next-line PhanUndeclaredFunction -- This is temp, pending pf4qpu-nc-p2
+		if ( ( function_exists( 'site_is_private' ) && site_is_private() ) || $is_coming_soon ) {
```

Comment:
> Probably just suppress this one for now, pending pf4qpu-nc-p2. Same for other functions from wpcomsh like `site_is_private()`.

Turns out I was wrong about `site_is_private()`. The call here is trying to use `\site_is_private()` (i.e. no namespace), while in wpcomsh the function has been `\Private_Site\site_is_private()` since it was added in https://github.com/Automattic/jetpack/commit/4237ebca2c3fc42a9352aadef0cc3f8d968a83d2.

Which I guess also means this has never worked for private sites and no one noticed? :sweat_smile:

Reply:
> Which I guess also means this has never worked for private sites and no one noticed?

@fushar @mmtr Could you please confirm the expectation here and if this is still relevant? Should I prepare a fix using `\Private_Site\site_is_private()` instead?

Reply:
This is working as expected on my Atomic site:

<img width="879" alt="Screenshot 2024-06-25 at 11 51 39" src="https://github.com/Automattic/jetpack/assets/1233880/c32fe07c-f873-4735-8aef-543559a9dce0">

I guess PHP is smart enough to know that `site_is_private` comes from the `Private_Site` namespace?

Reply:
Scratch that, the issue is specific to WP Admin:

<img width="966" alt="Screenshot 2024-06-25 at 11 53 42" src="https://github.com/Automattic/jetpack/assets/1233880/f76ca931-a2c1-490d-80dd-0fd3d1df3aa3">

Good catch, @coder-karen! Looks like this issue has been present since the first this was introduced in https://github.com/Automattic/jetpack/pull/17629.

> Should I prepare a fix using \Private_Site\site_is_private() instead?

That'd be great @fgiannar!

Reply:
Here's the fix (with an extra unit test for bonus points): https://github.com/Automattic/jetpack/pull/38026